examples/basic/ipc_shm_ipcapi_producer.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <umf/ipc.h>
+#include <umf/memory_pool.h>
+#include <umf/pools/pool_scalable.h>
+#include <umf/providers/provider_os_memory.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 1024
+
+int producer_connect(int port) {
+    struct sockaddr_in consumer_addr;
+    int producer_socket = -1;
+
+    // create a producer socket
+    producer_socket = socket(AF_INET, SOCK_STREAM, 0);
+    if (producer_socket < 0) {
+        fprintf(stderr, "[producer] ERROR: Unable to create socket\n");
+        return -1;
+    }
+
+    fprintf(stderr, "[producer] Socket created\n");
+
+    // set IP address and port the same as for the consumer
+    consumer_addr.sin_family = AF_INET;
+    consumer_addr.sin_port = htons(port);
+    consumer_addr.sin_addr.s_addr = inet_addr(INET_ADDR);
+
+    // send connection request to the consumer
+    if (connect(producer_socket, (struct sockaddr *)&consumer_addr,
+                sizeof(consumer_addr)) < 0) {
+        fprintf(stderr,
+                "[producer] ERROR: unable to connect to the consumer\n");
+        goto err_close_producer_socket_connect;
+    }
+
+    fprintf(stderr, "[producer] Connected to the consumer\n");
+
+    return producer_socket; // success
+
+err_close_producer_socket_connect:
+    close(producer_socket);
+
+    return -1;
+}
+
+int main(int argc, char *argv[]) {
+    char consumer_message[MSG_SIZE];
+    int producer_socket;
+    int ret = -1;
+
+    if (argc < 2) {
+        fprintf(stderr, "Usage: %s port\n", argv[0]);
+        return -1;
+    }
+
+    int port = atoi(argv[1]);
+
+    umf_memory_provider_handle_t OS_memory_provider = NULL;
+    umf_os_memory_provider_params_t os_params;
+    enum umf_result_t umf_result;
+
+    os_params = umfOsMemoryProviderParamsDefault();
+    os_params.visibility = UMF_MEM_MAP_SHARED;
+
+    // create OS memory provider
+    umf_result = umfMemoryProviderCreate(umfOsMemoryProviderOps(), &os_params,
+                                         &OS_memory_provider);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: creating OS memory provider failed\n");
+        return -1;
+    }
+
+    size_t page_size;
+    umf_result =
+        umfMemoryProviderGetMinPageSize(OS_memory_provider, NULL, &page_size);
```

Comment:
Why do we need to use `page_size`? I think the allocation size does not matter for the IPC example.

Reply:
Now I understand the reason. The `pool_proxy` implementation does not automatically round up the size to be multiple of the page size. But as I described in #475, I think the `pool_proxy` should be extended to support arbitrary allocation sizes.

Reply:
Removed

Reply:
A size can be any now, because we use `umfPoolCalloc(scalable_pool)` to allocate this memory.